test/declaration_test.ts

```diff
@@ -255,6 +255,39 @@ class X {
   // @Ignore
   X.fakeConstructor$();
   external x();
+}`);
+    });
+    it('should emit extension methods to return Futures from methods that return Promises', () => {
+      expectTranslate(`declare interface MyMath {
+        randomInRange(start: number, end: number): Promise<number>;
+      }`).to.equal(`import "dart:async" show Completer;
+
+@anonymous
+@JS()
+abstract class MyMath {
+  external dynamic /*Promise<num>*/ randomInRange(num start, num end);
+}
+
+extension on MyMath {
+  Future randomInRangeAsFuture(num start, num end) {
+    return _promiseToFuture(this.randomInRange(start, end));
+  }
+}
+
+Future<T> _promiseToFuture<T>(jsPromise) {
```

Comment:
add promiseToFuture to 
package:js/js_util.dart

Reply:
We have a copy in `dart:html`. We could expose it from `dart:js_util` as well...

Reply:
this would be a great fit for dart:js_util
We could also add more methods to js_util to make it fast to call JS methods with specific #s of arguments.